hpcbench/driver.py

```diff
@@ -977,6 +1003,13 @@ def _make_srun_arguments(self, optdict):
                 args.append('--{}'.format(k))
             else:
                 args.append('--{}={}'.format(k, six.moves.shlex_quote(str(v))))
+        if not isinstance(self.root.network.nodes(self.tag), ConstraintTag):
```

Comment:
So, if I understand this correctly, we ignore --constraint options in srun if the tag has it already set. I think that seems reasonable, but it would be good to throw an warning at the logger here because results could be quite confusing to the user.

Reply:
No. `srun` options are never ignored.
This change append `--nodelist` option to `srun` options if *constraint* is not already in `srun` options and if the tag is not a *constraint* one.

Reply:
ah, you're right!